src/core/namespace.c

```diff
@@ -204,6 +205,16 @@ static const MountEntry protect_control_groups_yes_table[] = {
         { "/sys/fs/cgroup",      MOUNT_READ_ONLY,         false  },
 };
 
+/* ProtectControlGroups=private table */
+static const MountEntry protect_control_groups_private_table[] = {
+        { "/sys/fs/cgroup",      MOUNT_PRIVATE_CGROUP2FS, false, .read_only = false, .nosuid = true, .noexec = true  },
```

Comment:
In the main namespace, we pass more mount options. See https://github.com/systemd/systemd/blob/f7078de515efd7089266815648b6e682f800649f/src/shared/mount-setup.c#L104-L109

Reply:
Ack had to add some complexity with condition_fn to check for memory_recursiveprot compatibility

Reply:
Never mind, just added memory_recursiveprot in mount_private_cgroup2fs like we do in mount_procfs for hidepid